### PR TITLE
Ian's code change done on Meraj's files

### PR DESCRIPTION
**Pull request by pyroib**
I have done an initial code change in Meraj's files. This will need a review before merging

### CODE REVIEW

File: src/FixBookUi.java [hunk 1]
```diff
@@ -1,78 +1,80 @@
...
+    
+    public FixBookUi(FixBookControl fixBookControl) {
+        this.fixBookControl = fixBookControl;
+        userInput = new Scanner(System.in);
+        uiState = UiState.INITIALISED;
+        fixBookControl.setUi(this);
+    }
+    
+    
+    public void setState(UiState state) {
```
Comment (jazzie-b) [inline src/FixBookUi.java]: Might want to change setState to setUiState to match variable name change, see #32

File: src/FixBookUi.java [hunk 2]
```diff
@@ -1,78 +1,80 @@
...
+        this.uiState = state;
+    }
+    
+    
+    public void run() {
```
Comment (jazzie-b) [inline src/FixBookUi.java]: Could use a more meaningful variable name, as requested. See #32

File: src/FixBookUi.java [hunk 3]
```diff
@@ -1,78 +1,80 @@
...
+    
+    
+    private String getUserInput(String prompt) {
+        System.out.print(prompt);
+        return userInput.nextLine();
+    }
+    
+    
+    public void printOutput(Object object) {
+        System.out.println(object);
```
Comment (jazzie-b) [inline src/FixBookUi.java]: Interesting change, very much valid.

File: src/FixBookUi.java [hunk 4]
```diff
@@ -1,78 +1,80 @@
...
+                            printOutput("Invalid bookId");
+                        }
+                    }
+                    break;
+                
+                case FIXING:
+                    String answer = getUserInput("Fix Book? (Y/N) : ");
+                    String uppercaseAnswer = answer.toUpperCase();
+                    boolean mustFix = false;
+                    if (uppercaseAnswer.equals("Y")) {
```
Comment (jazzie-b) [inline src/FixBookUi.java]: Still a method within method call, could be substituted using a boolean. See issue #32

File: src/Loan.java [hunk 5]
```diff
@@ -4,91 +4,103 @@
...
+        return member;
+    }
+    
+    
+    public Book getBook() {
+        return book;
+    }
+    
+    
+    public void dischargeLoan() {
```
Comment (jazzie-b) [inline src/Loan.java]: Not requested, but valid method name change

File: src/Loan.java [hunk 6]
```diff
@@ -4,91 +4,103 @@
...
+        return id;
+    }
+    
+    
+    public Date getDueDate() {
+        return date;
+    }
+    
+    
+    public String toString() {
```
Comment (jazzie-b) [inline src/Loan.java]: Method name ok, ignore request to change

File: src/Main.java [hunk 7]
```diff
@@ -1,224 +1,223 @@
...
+    private static void fixBooks() {
+        FixBookControl fixBookControl = new FixBookControl();
+        FixBookUi fixBookUi = new FixBookUi(fixBookControl);
+        fixBookUi.run();
+    }
+    
+    
+    private static void incrementDate() {
+        try {
+            int days = Integer.valueOf(userInput("Enter number of days: ")).intValue();
```
Comment (jazzie-b) [inline src/Main.java]: Still possibly a method within method call here, was requested to be looked. See #28

File: src/Main.java [hunk 8]
```diff
@@ -1,224 +1,223 @@
...
+            calender.incrementDate(days);
+            library.checkCurrentLoans();
+            printOutput(simpleDateFormat.format(calender.setDate()));
```
Comment (jazzie-b) [inline src/Main.java]: Another possible method with in method call, uncorrected. See #28

File: src/PayFineUi.java [hunk 9]
```diff
@@ -1,91 +1,96 @@
...
-	public void display(Object object) {
-		printOutput(object);
-	}
+    
+    
+    public static enum UiState {
+        INITIALISED, READY, PAYING, COMPLETED, CANCELLED
+    };
+    private PayFineControl payFineControl;
+    private Scanner input;
```
Comment (jazzie-b) [inline src/PayFineUi.java]: Variable name could be more meaningful as requested, see #30

**Review by jazzie-b — CHANGES_REQUESTED** [review]
Some additional changes may be require. See comments on review. Each comment has associated issue number attached.

File: src/FixBookUi.java (hunk 1 above)
Comment (pyroib) [reply src/FixBookUi.java]: done

File: src/FixBookUi.java (hunk 2 above)
Comment (pyroib) [reply src/FixBookUi.java]: done: runFixBookUi

**Comment by pyroib** [comment]
changes made. hopefully i got em all.

File: src/Main.java [hunk 10]
```diff
@@ -167,16 +167,18 @@ private static void returnBook() {
...
     
     private static void incrementDate() {
         try {
-            int days = Integer.valueOf(userInput("Enter number of days: ")).intValue();
+            String userInput = userInput("Enter number of days: ");
+            int days = Integer.valueOf(userInput).intValue();
             calender.incrementDate(days);
             library.checkCurrentLoans();
-            printOutput(simpleDateFormat.format(calender.setDate()));
+            String currentDate = simpleDateFormat.format(calender.setDate());
```
Comment (jazzie-b) [inline src/Main.java]: Still possible method within method call.

**Review by jazzie-b — COMMENTED** [review]
Might want to look at this one change. This branch is also going to need to be de-conflicted before merging.

**Comment by jazzie-b** [comment]
Branch has been deconflicted, one change still need to be made.

**Comment by pyroib** [comment]
that is now done. i found the same thing on line 36, changed that too

**Review by jazzie-b — APPROVED** [review]
All changes have been made. Approved to merge.

**Pull request merged**